### DISCUSSION TITLE
SideSetsFromNormalsGenerator produces speckled pattern instead of the full surface

### DISCUSSION THREAD

**Discussion by thiloschmid** [idaholab/moose#17359] [Q&A Meshing] [answered]
Dear all,

I'm a relatively new user of moose and for my first non-tutorial simulation I am stuck at defining sidesets in order to define boundary conditions. I have a mesh generated with gmsh that consists of a single surface. I wanted to use the **SideSetsFromNormalsGenerator** to simply have a separate sideset for the x and y side. If I only output the mesh the result is the speckled pattern seen below:

![sidesets](https://user-images.githubusercontent.com/28265235/111532659-31fcd800-8766-11eb-968b-49559466fadc.png)

My code looks like this:
    
    [Problem]
        solve = false
    []
    
    [Mesh]
        [roi]
            type = FileMeshGenerator
            file = 'mesh.msh'
            dim = 3
        []
    
        [sideSet]
            type = SideSetsFromNormalsGenerator
            input = roi
            new_boundary = 'x-side
                            y-side'
            replace = true
            normals = 
                '1 0 0
                 0 1 0'
            tolerance = 1e5
            variance = 1e-1
            fixed_normal = false
        []
    []
    
    [Outputs]
        exodus = true
    []

The goal would be to have the whole side covered. I tried playing around with the _tolerance_ and _variance_ parameters but without improving the result. The files I used are 

[sideset-problem.zip](https://github.com/idaholab/moose/files/6159481/sideset-problem.zip)

I also tried to use the **SideSetsFromPointsGenerator** but that did not produce any sideset at all.

Perhaps Moose is not the right tool for the job. However, e.g. Paraview does not offer a straight-forward way to define new sidesets or something similar. If you have an idea of any different approach or a different tool I could use, or if you think this is the wrong way to go proceed I would be happy to hear other suggestions. 

Thanks a lot!

**Comment by aeslaughter** [comment]
Boundaries such as this are typically added with the program that created the mesh. I am not familiar with gmsh, but perhaps @lindsayad could help out.

**Reply by thiloschmid** [reply]
Thanks for your comment.

Yes, I figured that there must be some other possibilities. I'm currently working with a single .stl file that encodes the full surface. In gmsh there is the possibility to assign different physical groups to geometric entities but at least in the GUI I could not find the option to split my one surface into multiple geometric entities in order to assign different names to different sections.

**Comment by GiudGiud** [comment]
The tolerance you are using on the sideset generator is too big. I think you meant to use 1e-5 not 1e5

**Reply by thiloschmid** [reply]
Changing the tolerance parameter was one of the first things I tried. At least visually, the values 1e5, 1, 1e-5 and 1e-10 all produce the same result.

**Comment by GiudGiud** [comment, marked as answer]
Closing this. 
I ll add a note to the sidesetfromnormalgenerator. It also does internal faces. 
The right one to use here was
https://mooseframework.inl.gov/source/meshgenerators/SideSetsAroundSubdomainGenerator.html
with a normals argument
